superset-frontend/src/SqlLab/components/QueryAutoRefresh/index.tsx

```diff
@@ -0,0 +1,109 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { useState } from 'react';
+import { SupersetClient } from '@superset-ui/core';
+import * as Actions from 'src/SqlLab/actions/sqlLab';
+import { Query } from 'src/SqlLab/types';
+import useInterval from 'src/SqlLab/utils/useInterval';
+
+const QUERY_UPDATE_FREQ = 2000;
+const QUERY_UPDATE_BUFFER_MS = 5000;
+const MAX_QUERY_AGE_TO_POLL = 21600000;
+const QUERY_TIMEOUT_LIMIT = 10000;
+
+interface UserOfflineFunc {
+  (offline: boolean): boolean;
+}
+
+interface RefreshQueriesFunc {
+  (alteredQueries: any): any;
+}
+
+interface Actions {
+  setUserOffline: UserOfflineFunc;
+  refreshQueries: RefreshQueriesFunc;
+}
+
+export interface QueryAutoRefreshProps {
+  queries: Query[];
+  actions: Actions;
+  queriesLastUpdate: number;
+}
+
+// returns true if the Query.state matches one of the specifc values indicating the query is still processing on server
+export const isQueryRunning = (q: Query): boolean =>
```

Comment:
@AAfghahi Agree it would be better to create a typescript enum for this in the types.ts file where Query is defined and use that. 

`/ / Possible states of a query object for processing on the server
export enum QueryState {
  STARTED = 'started',
  STOPPED = 'stopped',
  FAILED = 'failed',
  PENDING = 'pending',
  RUNNING = 'running',
  SCHEDULED = 'scheduled',
  SUCCESS = 'success',
  FETCHING = 'fetching',
  TIMED_OUT = 'timed_out',
}

// Indicates a Query's state is still processing
export const runningQueryStateList: QueryState[] = [
  QueryState.RUNNING,
  QueryState.STARTED,
  QueryState.PENDING,
  QueryState.FETCHING,
];

// Indictes a Query's state has completed processing regardless of success / failure
export const concludedQueryStateList: QueryState[] = [
  QueryState.STOPPED,
  QueryState.FAILED,
  QueryState.SUCCESS,
  QueryState.TIMED_OUT,
];

// check if a query is in a running state
export const isQueryRunning = (q: Query): boolean => runningQueryStateList.includes(q?.state);
`

 I have this setup and working on local but saw an additional query state value 'scheduled' in types.ts that seems like it should be included in the array used by isQueryRunning.  Anyone have context if 'scheduled' should be added to current list of ['running', 'started', 'pending', 'fetching']?

Reply:
Oh yeah I remember this, it is a bit confusing because I think some of these states only occur in the frontend and never get passed to the backend. I once broke this in superset, let me check to see what I needed to do to fix

Reply:
I think this is all of them: 
https://github.com/apache/superset/blob/master/superset-frontend/src/SqlLab/components/QueryTable/index.tsx#L117

Reply:
So I looked into that a bit, and it's related to Scheduled Queries. The config/feature doesn't seem to be working as expected, but regardless, let's assume that it will create a query with a status of schedule, so I'd say yes, let's go ahead and add that value as an option.

Reply:
@AAfghahi @eschutho Thank you for the context.  I just pushed up a commit that adds QueryState.SCHEDULED to runningQueryStateList